src/core/lib/iomgr/ev_poll_posix.cc

```diff
@@ -1437,9 +1444,24 @@ static void cache_destroy_locked(poll_args* args) {
     poll_cache.free_pollers = args->next;
   }
 
+  // Now move this args to the dead poller list for later join
+  if (poll_cache.dead_pollers != nullptr) {
+    poll_cache.dead_pollers->prev = args;
+  }
+  args->prev = nullptr;
+  args->next = poll_cache.dead_pollers;
   gpr_free(args);
```

Comment:
It looks like you were correct, we weren't doing refcounting for destroying ```args```, but I suspect we will have to start doing that to make these threads joinable.

Reply:
So is there anything else that can own args once it hits the freelist? I guess I'm not seeing the shared ownership here though I need to look more carefully.

Reply:
Never mind, I see it. Ok, we'll have to do something about it.